p2p/src/peer_manager/mod.rs

```diff
@@ -162,6 +251,19 @@ where
             P2pError::PeerError(PeerError::PeerAlreadyExists),
         );
 
+        if let (Some(receiver_address), Role::Outbound) = (receiver_address, role) {
+            self.handle_outbound_receiver_address(peer_id, receiver_address).await?;
+        }
+
+        if role == Role::Outbound {
```

Comment:
This place doesn't seem right. I'm trying to understand what's going on here. We should not make any communication before having completed the handshake. Is this done after the completion of the handshake and confirming that we can communicate with the peer? I see `validate_version` above, but I'm not sure about the order of things in both cases of inbound and outbound.

Reply:
Handshake exchange happens earlier in a different place (when peer event loop is started, for both inbound and outbound peers).

Reply:
Alrighty.